pkg/uploadserver/uploadserver.go

```diff
@@ -265,44 +265,62 @@ func (app *uploadServerApp) uploadHandler(w http.ResponseWriter, r *http.Request
 
 func newUploadStreamProcessor(stream io.ReadCloser, dest, imageSize, contentType string) error {
 	if contentType == FilesystemCloneContentType {
-		return filesystemCloneProcessor(stream, common.ImporterVolumePath)
-	}
-
-	if contentType == BlockdeviceCloneContentType {
```

Comment:
Jeepers, I can't believe this was an empty `if` block.  My intention was obviously to call `blockdeviceCloneProcessor()` which is now gone.  Should I not be surprised that the data processor is doing the right thing?  It's not going to use scratch space, right?

Reply:
It doing the right thing, and no it should not be using scratch space, since the pod didn't ask for it. Yeah I was boggled by the empty if, and started implementing it, but then I went, wait a minute, this works, we have tests that explicitly test this for block devices, and they work.